File: client/api/extras.ts
import { onPromise } from "@trippler/tr_lib/client"
import { checkVehicleExtras, toggleRemote, toggleCursor } from "../modules/extras"

const getVehicleClass = () => GetVehicleClass(GetVehiclePedIsIn(PlayerPedId(), false))

onPromise('getVehicleClass', getVehicleClass)

onNet('lenix_patrolextras:client:checkVehicleExtras', checkVehicleExtras)

onNet('lenix_patrolextras:client:toggleRemote', toggleRemote)
onNet('lenix_patrolextras:client:toggleCursor', toggleCursor)
exports('toggleRemote', toggleRemote)
exports('toggleCursor', toggleCursor)

export type VehicleClass = ReturnType<typeof getVehicleClass>

File: lenixm/lenix_patrol/client/api/extras.lua
local extrasMod<const> = require 'client/modules/extras'

local function getVehicleClass()
  return GetVehicleClass(GetVehiclePedIsIn(PlayerPedId(), false))
end

lib.onPromise('getVehicleClass', function()
  return getVehicleClass()
end)

RegisterNetEvent('lenix_patrolextras:client:checkVehicleExtras', function()
  extrasMod.checkVehicleExtras()
end)

RegisterNetEvent('lenix_patrolextras:client:toggleRemote', function()
  extrasMod.toggleRemote()
end)

RegisterNetEvent('lenix_patrolextras:client:toggleCursor', function()
  extrasMod.toggleCursor()
end)

exports('toggleRemote', extrasMod.toggleRemote)
exports('toggleCursor', extrasMod.toggleCursor)

return {
  getVehicleClass = getVehicleClass
}